pkg/agent/agent.go

```diff
@@ -140,11 +142,17 @@ func (i *Initializer) initInterfaceStore() error {
 			OFPort:   port.OFPort}
 		var intf *interfacestore.InterfaceConfig
 		switch {
-		case port.Name == i.hostGateway:
+		case port.OFPort == config.HostGatewayOFPort:
 			intf = &interfacestore.InterfaceConfig{
 				Type:          interfacestore.GatewayInterface,
 				InterfaceName: port.Name,
 				OVSPortConfig: ovsPort}
+			if intf != nil && intf.InterfaceName != i.hostGateway {
```

Comment:
why it checks `intf != nil`? it's impossible to be false in the context

Reply:
Right. Removed it. Thanks for catching it.